### DISCUSSION TITLE
A Pie Chart to show distribution of incomming quries by protocol

### DISCUSSION THREAD

**Discussion by rehanone** [TechnitiumSoftware/DnsServer#389] [Ideas]
First of all, a million thanks for creating such a useful open source tool. I have set it up on multiple cloud cloud providers as a redundant DNS backend for my homelab and it has been a great experience so far with the exception of a few minor bugs (Which I will report separately). Since I am using both HTTPS and TLS on my servers, I was interested in knowing what is the distribution of incoming queries look like. E.g., HTTPS, TLS, TCP, UDP. I can see this information in the Query Logs (sqlite) app but I was wondering if there is a way to show that as a pie-chart on the main dashboard?

Thanks once again for all your hard work!

**Comment by ShreyasZare** [comment]
Thanks for the compliments. The data for protocol usage is not stored in the stats data set so its not directly available to use on dashboard. The Query Log app is independent and the DNS server does not depend on it for the dashboard stats. The stats data is tied with DNS server's internal features like cache prefetching, query limiting, etc. so cannot keep on adding more data points in it asit will take up large amount of memory.

Probably it will be good to add more advance views that use the app as the data source.
